webui/react/src/pages/ResourcePool/ResourcepoolDetail.tsx

```diff
@@ -219,11 +224,7 @@ const ResourcepoolDetailInner: React.FC = () => {
             size={ShirtSize.Large}
           />
         </Section>
-        {!!agents.length && poolname && (
-          <Topology
-            nodes={agents.filter(({ resourcePools }) => resourcePools.includes(poolname))}
-          />
-        )}
+        {!!topologyAgentPool.length && poolname && <Topology nodes={topologyAgentPool} />}
```

Comment:
**question**: this looks better. I'm not too familiar here, but it's kind of odd that the `ResourcepoolDetail.tsx` doesn't make `poolname` required and is optional. Is there a case for the card to handle `undefined` value?